dyno/panda/dyno_panel.py

```diff
@@ -0,0 +1,133 @@
+import tkinter as tk
+from tkinter import font
+from PIL import Image, ImageTk
+import serial
+import time
+
+#Define constants
+NUMSENSORS1 = 2 #RPM AND TEMPERATURE
+NUMSENSORS2 = 2 #JOULEMETER
+#---------------------
+
+#Images
+dino_path = "./img/dino.gif"
+start_path = "./img/start.png"
+#---------------------
+
+#Colors
+color_bg = "#16495C"
+color_fg = "#DA8669"
+color_label = "#B1D9DB"
+#---------------------
+
+#Placement
+LABEL_COL1_PLACE = {'relx':0.15, 'rely':0.6}
+LABEL_COL2_PLACE = {'relx':0.55, 'rely':0.6}
+LABEL_SPACE = 0.08
+#---------------------
+
+#Open the serial port that the MCU is connected to
+'''ser = serial.Serial('/dev/ttyUSB0', 9600, timeout=0.5)'''
+#---------------------
+
+#Update the variables for the labels
+def update_labels():
+    #Read in input from MCU sensors
+    '''for x in range (0, NUMSENSORS1): #read sensor statements
+        val = ser1.readline().decode('ascii')
+        prefix = val[0:3]
+        my_dict[prefix] = float(val[5:-2])
+    for x in range (0, NUMSENSORS2): #read JOULEMETER statements
+        val = ser2.readline().decode('ascii')
+        prefix = val[0:3]
+        my_dict[prefix] = float(val[5:-2])'''
+    for key in my_dict.keys():
```

Comment:
Hmm I see what you mean. I would still prefer the code that lands on master to be production-ready though. Maybe there's a compromise here with something like:
```suggestion
    if serial_read_on:   
        for x in range (0, NUMSENSORS1): #read sensor statements
            val = ser1.readline().decode('ascii')
            prefix = val[0:3]
            my_dict[prefix] = float(val[5:-2])
        for x in range (0, NUMSENSORS2): #read JOULEMETER statements
            val = ser2.readline().decode('ascii')
            prefix = val[0:3]
            my_dict[prefix] = float(val[5:-2])
    for key in my_dict.keys():
```
You'd have to add support for that `serial_read_on` flag though, probably by having it on by default and allowing the user to disable it when running something like:
`python dyno_panel.py --no-serial`

What are your thoughts on this?

Reply:
Thats absolutely brilliant. Will definitely add